tests/framework/results/test_interface.py

```diff
@@ -352,34 +372,35 @@ def test_unhashable_pipeline(mocker):
             pop_filter='alive == "alive" and undead == False',
             when="collect_metrics",
             requires_columns=[],
-            requires_values=[["bad", "unhashable", "thing"]],  # unhashable first element
+            # Unhashable first element below
+            requires_values=[["bad", "unhashable", "thing"]],  # type: ignore[list-item]
             additional_stratifications=[],
             excluded_stratifications=[],
             aggregator_sources=[],
             aggregator=_silly_aggregator,
         )
 
 
-def mock__prepare_population(self, event):
-    """Return a mock population in the vein of ResultsManager._prepare_population"""
-    # Generate population DataFrame
-    population = BASE_POPULATION.copy()
-
-    # Mock out some extra columns that would be produced by the manager's _prepare_population() method
-    population["current_time"] = pd.Timestamp(year=2045, month=1, day=1, hour=12)
-    population["event_step_size"] = timedelta(days=28)
-    population["event_time"] = pd.Timestamp(year=2045, month=1, day=1, hour=12) + timedelta(
-        days=28
-    )
-    return population
-
-
 @pytest.mark.parametrize(
     "when",
     ["time_step__prepare", "time_step", "time_step__cleanup", "collect_metrics"],
 )
-def test_register_adding_observation_when_options(when, mocker):
+def test_register_adding_observation_when_options(when: str, mocker: MockerFixture) -> None:
     """Test the full interface lifecycle of adding an observation and simulation event."""
+
+    def mock__prepare_population(self: ResultsManager, event: Event) -> pd.DataFrame:
```

Comment:
Why was this necessary?

Reply:
It wasn't, I just like it in local scope since no other tests were using it (it seems to be how some/most? of the mock functions are handled throughout the repo)